addon/internal-session.js

```diff
@@ -17,25 +17,24 @@ export default Ember.ObjectProxy.extend(Ember.Evented, {
     Ember.assert(`Session#authenticate requires the authenticator to be specified, was "${authenticator}"!`, !Ember.isEmpty(authenticator));
     let theAuthenticator = getOwner(this).lookup(authenticator);
     Ember.assert(`No authenticator for factory "${authenticator}" could be found!`, !Ember.isNone(theAuthenticator));
-    return new Ember.RSVP.Promise((resolve, reject) => {
+    return new RSVP.Promise((resolve, reject) => {
       theAuthenticator.authenticate.apply(theAuthenticator, args).then((content) => {
-        this._setup(authenticator, content, true);
-        resolve();
+        this._setup(authenticator, content, true).then(resolve, reject);
       }, (error) => {
-        this._clear();
-        reject(error);
+        this._clear().finally(() => {
+          reject(error);
+        });
```

Comment:
You are right, there is a difference, but is this undesired?
The `finally` method pollutes the list a bit because it adds an extra fulfilling promise.

```javascript
Ember.RSVP.Promise.reject().then(function(){
  console.log('resolve');
}, function(){
  console.log('reject');
});
```
![schermafbeelding 2015-11-30 om 22 14 50](https://cloud.githubusercontent.com/assets/2470404/11484786/5ac1107c-97b0-11e5-8618-bcbb7d9a0079.png)

```javascript
Ember.RSVP.Promise.reject().finally(function(){
  console.log('finally');
});
```
![schermafbeelding 2015-11-30 om 22 15 42](https://cloud.githubusercontent.com/assets/2470404/11484771/3c96c092-97b0-11e5-9c8b-4435bc79490f.png)

```javascript
Ember.RSVP.Promise.reject().then(function(){
  console.log('resolve');
}, function(){
  console.log('reject');
}).finally(function(){
  console.log('finally');
});
```
![schermafbeelding 2015-11-30 om 22 19 31](https://cloud.githubusercontent.com/assets/2470404/11484815/7bf90920-97b0-11e5-8073-b4baac4c3048.png)